specs/clipboard-api/clipboard-api.md

```diff
@@ -0,0 +1,337 @@
+    formatting stripped out. Many applications let you do this, but some
+    don't, especially when the app from which the formatted text was cut
+    or copied only provides the formatted version to the clipboard APIs.
+    We could implement a "to plain text" feature to automatically create
+    plain text from formatted HTML or RTF text when needed on either cut/copy
+    or paste, then update the edit controls in Microsoft.UI.Xaml
+    to use the Reunion clipboard API and therefore take advantage of
+    this feature.
+
+  * _Clipboard history:_ Windows 10 version 1809 (October/November 2018 update)
+    introduced a new built-in clipboard history manager. The Project Reunion SDK
+    could allow us to bring the clipboard history manager to older Windows versions.
+    To do this, we might conceivably use the same "main
+    package" as the [Project Reunion brokering system](
+    ../../docs/README.md#brokering) to install a Windows process
+    to host the shared memory for the clipboard history.
+    Then, the APIs and user interface for clipboard history
+    could be ported to work on that main package process
+    rather than the code shipping in Windows.
+    (We could also consider using [PowerToys](
+      https://github.com/microsoft/PowerToys) as the delivery vehicle
+    for the clipboard history UI, while keeping the shared memory store
+    and API implementations as part of the Reunion SDK.)
+  
+  * _Cloud clipboard sync and roaming:_ Windows 10 version 1809 also
+    introduced cloud clipboard sync to other Windows devices you sign in to
+    with the same personal Microsoft account or work or school Azure
+    Active Directory account. Just as with clipboard history,
+    we could use use the Reunion SDK main package
+    to bring the cloud clipboard sync client to older Windows versions.
+  
+  * _Copy append/paste multiple_: We've also seen several suggestions from
+    Windows users for a way to cut or copy multiple separate items,
+    then paste them all at once in the same order that they were
+    originally cut or copied. If we did implement this feature,
+    we could put its memory store in the Reunion SDK main package
+    to bring it to older Windows releases.
+
+## Description
+
+When you use Windows' built-in user interface controls, you get their
+built-in support for copying and pasting data for free. However,
+when you make your own controls or use controls from a third-party vendor,
+you might need to implement copying and pasting yourself.
+You can do this with the clipboard API in the Project Reunion SDK.
+
+The Reunion SDK clipboard API is very similar to Windows' built-in
+"UWP clipboard" API in the [Windows.ApplicationModel.DataTransfer](
+https://docs.microsoft.com/uwp/api/windows.applicationmodel.datatransfer)
+Windows Runtime namespace.
+In fact, in this initial version (as of mid-2020), the Reunion
+SDK clipboard API is the same as Windows.ApplicationModel.DataTransfer,

```

Comment:
This does mean that we import the asynchronous model of most UWP classes, like the ones in Windows.ApplicationModel.DataTransfer. @usvoyager suggested at one point we should review the async behavior for usability. I forget what he was referring to but I think it had to do with the inconsistency of sync versus async APIs in DataTransfer.